WarshipGame/Assets/Scripts/Gameplay/Grid/GlowManager.cs

```diff
@@ -0,0 +1,118 @@
+using System.Collections.Generic;
+using UnityEngine;
+
+/// <summary>
+/// Manages material colors and shaders to be activated and changed on selection.
+/// </summary>
+public class GlowManager : MonoBehaviour
+{
+    private readonly Dictionary<Renderer, Material[]> _selectionMaterialDictionary = new Dictionary<Renderer, Material[]>();
+    private readonly Dictionary<Renderer, Material[]> _originalMaterialDictionary = new Dictionary<Renderer, Material[]>();
+    private readonly Dictionary<Color, Material> _cachedSelectionMaterials = new Dictionary<Color, Material>();
+    private readonly Color _validHexColor = Color.green;
+    private Color _originalColor;
+    
+    public Material SelectionMaterial;
+    public bool IsGlowing = false;
+    
+    private void Awake()
+    {
+        PrepareMaterialDictionaries();
+        _originalColor = SelectionMaterial.GetColor("_GlowColor");
+    }
+
+    /// <summary>
+    /// Fills in material dictionaries for later use 
+    /// </summary>
+    private void PrepareMaterialDictionaries()
+    {
+        foreach (Renderer renderer in GetComponentsInChildren<Renderer>())
+        {
+            Material[] originalMaterials = renderer.materials;
+            _originalMaterialDictionary.Add(renderer, originalMaterials);
+            Material[] newMaterials = new Material[renderer.materials.Length];
+
+            for (int i = 0; i < originalMaterials.Length; i++)
+            {
+                Material mat = null;
+                if (!_cachedSelectionMaterials.TryGetValue(originalMaterials[i].color, out mat))
+                {
+                    mat = new Material(SelectionMaterial)
+                    {
+                        color = originalMaterials[i].color
+                    };
+                    _cachedSelectionMaterials.Add(originalMaterials[i].color, mat);
+                }
+
+                newMaterials[i] = mat;
+            }
+
+            _selectionMaterialDictionary.Add(renderer, newMaterials);
+        }
+    }
+
+    /// <summary>
+    /// Toggles the glow on or off when the function is called
+    /// </summary>
+    public void ToggleGlow()
+    {
+        if (!IsGlowing)
+        {
+            foreach (Renderer renderer in _originalMaterialDictionary.Keys)
+            {
+                renderer.materials = _selectionMaterialDictionary[renderer];
+            }
+            IsGlowing = true;
+            return;
+        }
+
+        foreach (Renderer renderer in _originalMaterialDictionary.Keys)
+        {
+            renderer.materials = _originalMaterialDictionary[renderer];
+        }
+
+        IsGlowing = false;
+    }
+
+    /// <summary>
+    /// Returns the state of the glow when the function is called
+    /// </summary>
+    /// <param name="state"></param>
+    public void ToggleGlow(bool state)
+    {
+        if (IsGlowing == state) return;
+        IsGlowing = !state;
```

Comment:
`!state` can be changed into `!IsGlowing` for easier understanding of what this is doining